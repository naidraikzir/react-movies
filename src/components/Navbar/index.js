import React from 'react';
import { useHistory } from 'react-router-dom';
import {
  Flex,
  Heading,
  Spacer
} from '@chakra-ui/react';
import { ColorModeSwitcher } from 'components/ColorModeSwitcher';

const Navbar = ({
  onTitleClick,
}) => {
  const history = useHistory();

  return (
    <Flex p="4">
      <Heading
        cursor="pointer"
        ml="1"
        onClick={() => history.push('/')}
      >
        Movies
      </Heading>
      <Spacer />
      <ColorModeSwitcher justifySelf="flex-end" />
    </Flex>
  );
};

export default Navbar;
