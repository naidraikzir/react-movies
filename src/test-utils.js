import React from 'react';
import { render } from '@testing-library/react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { ChakraProvider, theme } from '@chakra-ui/react';
import reducer from 'store/reducer';

const customRender = (
  ui,
  {
    initialState,
    store = createStore(reducer, initialState),
    ...options
  } = {}
) => {
  const AllProviders = ({ children }) => (
    <Provider store={store}>
      <ChakraProvider theme={theme}>{children}</ChakraProvider>
    </Provider>
  );
  return render(ui, { wrapper: AllProviders, ...options });
}

export { customRender as render };
