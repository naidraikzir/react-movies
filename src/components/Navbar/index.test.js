import React from 'react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import { fireEvent } from '@testing-library/react';
import { render } from 'test-utils';
import Navbar from '.';

test('match snapshot', () => {
  const { container } = render(<Navbar />);
  expect(container.firstChild).toMatchSnapshot();
});

test('navigate to root on title click', () => {
  const history = createMemoryHistory();
  const { getByText } = render(<Router
    history={history}
  >
    <Navbar />
  </Router>);
  history.push('/somewhere');
  fireEvent.click(getByText('Movies'));
  expect(history.location.pathname).toEqual('/');
});
