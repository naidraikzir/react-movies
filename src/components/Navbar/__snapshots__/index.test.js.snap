// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  class="css-nc95w2"
>
  <h2
    class="chakra-heading css-567jj8"
  >
    Movies
  </h2>
  <div
    class="css-17xejub"
  />
  <button
    aria-label="Switch to dark mode"
    class="chakra-button css-4vahhp"
    type="button"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      focusable="false"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
      />
    </svg>
  </button>
</div>
`;
